src/solvers/flattening/bv_pointers.cpp

```diff
@@ -203,6 +204,10 @@ literalt bv_pointerst::convert_rest(const exprt &expr)
           bv_utilst::representationt::SIGNED));
     }
   }
+  else if(expr.id() == ID_r_ok || expr.id() == ID_w_ok || expr.id() == ID_rw_ok)
+    return convert(simplify_expr(to_r_or_w_ok_expr(expr).lower(ns), ns));
```

Comment:
It seems we don't have any test that triggers this lowering step. Do we really get the guarantee that rw_ok expressions are going to be converted only when going to the solver ? (assuming this is what this PR tries to achieve ?)

Reply:
Should now be done, the early-expansion removal wasn't there back then.